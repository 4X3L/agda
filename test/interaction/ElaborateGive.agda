module ElaborateGive where

open import Agda.Builtin.List
open import Agda.Builtin.Nat
open import Agda.Builtin.Reflection
open import Agda.Builtin.Unit

macro
  addUnknown : Term → TC ⊤
  addUnknown goal = unify goal
                          (def (quote _+_)
                            (arg (arg-info visible relevant) unknown ∷
                            arg (arg-info visible relevant) unknown ∷ []))

unknownPlus : Nat
unknownPlus = {!!}

five : Nat
five = {!!}
