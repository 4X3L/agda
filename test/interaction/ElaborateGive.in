top_command (cmd_load currentFile [])
goal_command 0 (cmd_elaborate_give Simplified) "addUnknown"
goal_command 1 (cmd_elaborate_give Simplified) "2 + 3"
