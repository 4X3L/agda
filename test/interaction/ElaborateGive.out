(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : Nat ?1 : Nat " nil)
((last . 1) . (agda2-goals-action '(0 1)))
(agda2-give-action 0 "? + ?")
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?1 : Nat ?2 : Nat ?3 : Nat " nil)
((last . 1) . (agda2-goals-action '(2 3 1)))
(agda2-give-action 1 "5")
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?2 : Nat ?3 : Nat " nil)
((last . 1) . (agda2-goals-action '(2 3)))
